rpc/interface/src/route/bin.rs

```diff
@@ -66,8 +73,16 @@ macro_rules! generate_endpoints_inner {
     ($variant:ident, $handler:ident, $request:expr) => {
         paste::paste! {
             {
+                // Check if restricted.
+                if [<$variant Request>]::IS_RESTRICTED && $handler.restricted() {
+                    // TODO: mimic `monerod` behavior.
+                    return Err(StatusCode::FORBIDDEN);
+                }
```

Comment:
Unrelated: binary requests now check `restricted`.

All the binary requests are `!restricted` but adding this for the future is probably a good idea.